Issuance: c_nonce and jwt proof clarification (#208)
italia/eidas-it-wallet-docs: docs/en/pid-eaa-issuance.rst

**Steps 16-17 (DPoP Proof for Credential Endpoint):** The Wallet Instance for requesting the Digital Credential creates a proof of possession with ``c_nonce`` obtained in **Step 15** and using the private key used for the DPoP, signing a DPoP Proof JWT according to (:rfc:`9449`) Section 4. The ``jwk`` value in the ``proof`` parameter MUST be equal to the public key referenced in the DPoP.

> **Step 18 (Credential Request):** The Wallet Instance requests the Digital Credential to the PID/(Q)EAA Credential endpoint. The request MUST contain the *Access Token*, the *DPoP Proof JWT*, the *credential type*, the `proof` (proof of possession of the key), and the ``format`` parameters, where the *proof* is an object containing the proof of possession of the cryptographic key material the issued PID/(Q)EAA Digital Credential would be bound to. The PID/(Q)EAA Provider performs the following checks on the Credential endpoint to verify the *proof*:

Reviewer: ```suggestion
**Step 18 (Credential Request):** The Wallet Instance sends a request for the Digital Credential to the PID/(Q)EAA Credential endpoint. This request MUST include the Access Token, DPoP Proof JWT, credential type, proof (which demonstrates possession of the key), and format parameters. The proof parameter MUST be an object that contains evidence of possession of the cryptographic key material to which the issued PID/(Q)EAA Digital Credential will be bound. To verify the proof, the PID/(Q)EAA Provider conducts the following checks at the Credential endpoint:
```

  1. all required claims for the *JWT proof* type are included as defined in the table of Section :ref:`Token Request <sec_token_request>`; 
  2. the key proof is explicitly typed using header parameters as defined for that *proof* type; 
  3. the header parameter indicates a registered asymmetric digital signature algorithm, alg parameter value MUST not be set to *none*; 
  4. the signature on the key proof is verified with the public key contained in the header parameter;  
  5. the header parameter does not contain a private key;

>   6. the *nonce* claim matches the server-provided *c_nonce* value, if the server had previously provided a *c_nonce*, the creation time of the JWT, as determined by either the issuance time (*iat*), or a server-managed timestamp via the *nonce* claim, is within an acceptable window. 

Reviewer: 1. the JWT proof MUST include all required claims as specified in the table of Section :ref:`Token Request <sec_token_request>`;
2. The key proof MUST be explicitly typed using header parameters as defined for the respective proof type;
3. The header parameter alg MUST indicate a registered asymmetric digital signature algorithm, and MUST NOT be set to `none`;
4. The signature on the key proof MUST be verified using the public key specified in the header parameter.
5. The header parameter MUST NOT contain a private key.
6. If a `c_nonce` value was previously provided by the server, the nonce claim in the JWT MUST match this `c_nonce` value. Furthermore, the creation time of the JWT, as indicated by the `iat` claim or a server-managed timestamp via the nonce claim, MUST be within an acceptable window of time as determined by the server.

>   1. all required claims for the *JWT proof* type are included as defined in the table of Section :ref:`Token Request <sec_token_request>`; 
>   2. the key proof is explicitly typed using header parameters as defined for that *proof* type; 
>   3. the header parameter indicates a registered asymmetric digital signature algorithm, alg parameter value MUST not be set to *none*; 
>   4. the signature on the key proof is verified with the public key contained in the header parameter;  
>   5. the header parameter does not contain a private key;
>   6. the *nonce* claim matches the server-provided *c_nonce* value, if the server had previously provided a *c_nonce*, the creation time of the JWT, as determined by either the issuance time (*iat*), or a server-managed timestamp via the *nonce* claim, is within an acceptable window. 

Reviewer: ```suggestion
 1. the JWT proof MUST include all required claims as specified in the table of Section :ref:`Token Request <sec_token_request>`;
 2. The key proof MUST be explicitly typed using header parameters as defined for the respective proof type;
 3. The header parameter alg MUST indicate a registered asymmetric digital signature algorithm, and MUST NOT be set to `none`;
 4. The signature on the key proof MUST be verified using the public key specified in the header parameter.
 5. The header parameter MUST NOT contain a private key.
 6. If a `c_nonce` value was previously provided by the server, the nonce claim in the JWT MUST match this `c_nonce` value. Furthermore, the creation time of the JWT, as indicated by the `iat` claim or a server-managed timestamp via the nonce claim, MUST be within an acceptable window of time as determined by the server.
```
